File: go/462_MinimumMovesToEqualArrayElementsII.go
package main

import "sort"

func minMoves2(nums []int) int {
    // 本质就是使分配在某个点左右两边的数量相等或差值为1
    sort.Ints(nums)
    target := nums[len(nums)/2]
    moves := 0
    for _, v := range nums {
        if v > target {
            moves += v - target
        } else {
            moves += target - v
        }
    }
    return moves
}

File: python/462_MinimumMovesToEqualArrayElementsII.py
from typing import List


class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        # 本质就是使分配在某个点左右两边的数量相等或差值为1
        nums.sort()
        target = nums[len(nums) // 2]
        return sum(abs(target - v) for v in nums)
